lib/services/database.dart

```diff
@@ -10,18 +10,32 @@ class DatabaseService {
   final CollectionReference userCollection =
       FirebaseFirestore.instance.collection('students');
 
-  Future<void> updateStudent(String name, String yearId) async {
-    return await userCollection.doc(uid).set({
-      'name': name,
-      'yearId': yearId,
+  Future<void> setStudent({String name, String yearId}) async {
+    return await userCollection
+        .doc(uid)
+        .set({'name': name, 'yearId': yearId, 'favoriteTAs': []});
+  }
+
+  Future<void> updateStudent(
+      {String name, String yearId, List<String> favoriteTAs}) async {
+    return await userCollection.doc(uid).update({
+      if (name != null) 'name': name,
+      if (yearId != null) 'yearId': yearId,
+      if (favoriteTAs != null) 'favoriteTAs': favoriteTAs
     });
   }
 
   Student _studentFromSnapshot(DocumentSnapshot snapshot) {
-    return Student(uid, snapshot.data()['name'], snapshot.data()['yearId']);
+    return Student(
```

Comment:
Should check if the values are null and provide default values (like empty string or empty arrays) or an error.

Reply:
these couldn't be null because of validation on the form they are gotten from

Reply:
The db might old data and the structure of the data or class might change. This code would not be backward or forward compatible. So default checks would be nice.

Reply:
Well, ok, I'll fix it. Quick reminder - I remember that once I've placed a comment with the same idea to you, but your answer was mostly like "it can't break in current realization" and for me it was ok🙂

Reply:
added